### PR TITLE
Add trychooser support

### PR DESCRIPTION
**Pull request by Manishearth**
Depends on https://github.com/servo/homu/pull/167

homu SQLite migration notes: Either start from a fresh db, or run `ALTER TABLE pull ADD try_choose text`

r? @jdm

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/873)

### CODE REVIEW

File: buildbot/master/files/config/master.cfg [hunk 1]
```diff
@@ -60,6 +60,13 @@ def servo_master_filter(c):
             c.who.startswith('bors-servo') and
             c.branch == "master")
 
+def servo_try_chooser_filter(f):
+    branch = "try-%s" % f
```
Comment (aneeshusa) [inline buildbot/master/files/config/master.cfg]: nit: inline this

File: homu/map.jinja [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {%
   set homu = {
-    'db': '/var/homu/main.db',
-    'rev': '992e774d7144e810193bcaede827339663c1301a'
+    'db': '/var/homu/main_test.db',
+    'rev': 'ca6ba9605d6506edc74b98f95cdac62bbfc5deea',
+    'servo_try_choosers': {
+        build': ['linux-dev'],
+        'wpt': ['linux-rel-css', 'linux-rel-wpt']
+        'android': ['android']
```
Comment (jdm) [inline homu/map.jinja]: Let's add android-x86.

File: homu/files/cfg.toml [hunk 3]
```diff
@@ -240,6 +240,11 @@ try_builders = [
 username = "{{ secrets["buildbot-http-user"] }}"
 password = "{{ secrets["buildbot-http-pass"] }}"
 
+{% for chooser, builders in homu.try_choosers.items() %}
+[repo.servo.buildbot."{{ chooser }}"]
```
Comment (jdm) [inline homu/files/cfg.toml]: Shouldn't this be `buildbot.try_choosers."{{ chooser }}"`?

**Comment by Manishearth** [comment]
Done.

**Comment by Manishearth** [comment]
The array doesn't format quite right so I explicitly printed it out instead of relying on the stringifier.

**Comment by Manishearth** [comment]
It now seems to work on vagrant

**Comment by jdm** [comment]
```
[ FAIL ] Buildbot master config lint check failed:
         ./buildbot/master/files/config/master.cfg:63:1: E302 expected 2 blank lines, found 1
         ./buildbot/master/files/config/master.cfg:96:2: E225 missing whitespace around operator
         ./buildbot/master/files/config/master.cfg:96:3: E901 SyntaxError: invalid syntax
         ./buildbot/master/files/config/master.cfg:96:62: E225 missing whitespace around operator
         ./buildbot/master/files/config/master.cfg:101:80: E501 line too long (89 > 79 characters)
         ./buildbot/master/files/config/master.cfg:103:2: E225 missing whitespace around operator
         ./buildbot/master/files/config/master.cfg:103:12: E225 missing whitespace around operator
```

**Comment by jdm** [comment]
```
[ FAIL ] Buildbot master config lint check failed:
         home/servo/buildbot/master/master.cfg:1:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'home/servo/buildbot/master/master.cfg'
```

File: tests/sls/buildbot/master/config_lint.py [hunk 4]
```diff
@@ -13,7 +13,7 @@ def run():
         'config'
     )
     # Have to specify master.cfg separately because it is not a .py file
-    command = ['flake8', CONF_DIR, os.path.join(CONF_DIR, 'master.cfg')]
+    command = ['flake8', CONF_DIR, "home/servo/buildbot/master/master.cfg"]
```
Comment (jdm) [inline tests/sls/buildbot/master/config_lint.py]: Revert this change?

Comment (Manishearth) [reply tests/sls/buildbot/master/config_lint.py]: No, this is intentional, this lets us lint the generated file.

The other option is to locally jinja it but I tried it and the salt jinja format is weird so I'd have to convert first.
